pyserini/search/lucene/_impact_searcher.py

```diff
@@ -53,19 +53,23 @@ class LuceneImpactSearcher:
         QueryEncoder to encode query text
     """
 
-    def __init__(self, index_dir: str, query_encoder: Union[QueryEncoder, str], min_idf=0):
+    def __init__(self, index_dir: str, query_encoder: Union[QueryEncoder, str], min_idf=0, onnx_encoder: Union[QueryEncoder, str]=None):
```

Comment:
hrm... not sure if I like this design, since the encoders are separate parameters... maybe we can have `encoder_name` and `encoder_type`?